### ISSUE TITLE
[no-use-before-declare] false positive for variables used from inner function scopes

### ISSUE THREAD

**Issue by csnover** [palantir/tslint#199]
Given the following simplified test case:

``` ts
var a = function ():void {
    b();
}

var b = function ():void {
    a();
};
```

Actual: Warning: “variable 'b' used before declaration, line 2 column 5”
Expected: No warning since it is not being used until `a` is called

tslint version 0.4.12
